src/LevelDBStore/Comparator.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2015-2024 The Neo Project.
+//
+// Comparator.cs file belongs to the neo project and is free
+// software distributed under the MIT software license, see the
+// accompanying file LICENSE in the main directory of the
+// repository or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using LevelDB.NativePointer;
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace LevelDB
+{
+    public class Comparator : LevelDBHandle
+    {
+        private sealed class Inner : IDisposable
+        {
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+            private delegate void Destructor(IntPtr gCHandleThis);
+
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+            private delegate int Compare(IntPtr gCHandleThisg,
+                                         IntPtr data1, IntPtr size1,
+                                         IntPtr data2, IntPtr size2);
+
+            [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+            private delegate IntPtr Name(IntPtr gCHandleThis);
+
+
+            private static readonly Destructor s_destructor
+                = (gCHandleThis) =>
+                      {
+                          var h = GCHandle.FromIntPtr(gCHandleThis);
+                          var @this = h.Target as Inner;
+
+                          @this.Dispose();
+
+                          // TODO: At the point 'Free' is entered, this delegate may become elligible to be GC'd.
+                          // TODO:  Need to look whether GC might run between then, and when this delegate returns.
+                          h.Free();
+                      };
+
+            private static readonly Compare s_compare =
+                (gCHandleThis, data1, size1, data2, size2) =>
+                    {
+                        var @this = GCHandle.FromIntPtr(gCHandleThis).Target as Inner;
+                        return @this._cmp(new NativeArray { _baseAddr = data1, _byteLength = size1 },
+                                        new NativeArray { _baseAddr = data2, _byteLength = size2 });
+                    };
+
+            private static readonly Name s_nameAccessor =
+                (gCHandleThis) =>
+                    {
+                        var @this = GCHandle.FromIntPtr(gCHandleThis).Target as Inner;
+                        return @this.NameValue;
+                    };
+
+            private Func<NativeArray, NativeArray, int> _cmp;
+            private GCHandle _namePinned;
+
+            public IntPtr Init(string name, Func<NativeArray, NativeArray, int> cmp)
+            {
+                // TODO: Complete member initialization
```

Comment:
We should finish the code

Reply:
It's there if you need it.

@shargon i reverted all your changes. Please don't make changes to my code directly unless you have good reason. You change my code all the time to leave it broken or add bugs and you don't understand the reason behind the design.